oecophylla/cli/launch.py

```diff
@@ -86,67 +122,97 @@ def _create_dir(_path):
               help='Restarts the run and overwrites previous input.')
 @click.option('--just-config', is_flag=True, flag_value=False,
               help='Only generate the configuration file.')
+@click.option('--test', is_flag=True, default=False,
+              help='Executes a run with the included test data.')
 def workflow(targets, input_dir, sample_sheet, params, envs,
              cluster_config, local_scratch, workflow_type, output_dir,
              snakemake_args, local_cores, jobs,
-             force, just_config):
+             force, just_config, test):
     import snakemake
     from skbio.io.registry import sniff
 
     # SNAKEMAKE
     snakefile = "%s/../../Snakefile" % os.path.abspath(os.path.dirname(__file__))
 
-    # INPUT DIR
-    for inp_file in glob.glob('%s/*' % input_dir):
-
-        file_format = sniff(inp_file)[0]
-
-        if (file_format != 'fasta') and (file_format != 'fastq'):
-            raise TypeError('Input files need to be in FASTA or FASTQ format.')
-
-    # OUTPUT
-    # create output directory, if does not exist
-    _create_dir(output_dir)
-
-    # SAMPLE SHEET
-    # If a manifest is not specified, the files containing forward and reverse reads
-    # will be automatically identified using regex.
-    if sample_sheet:
-        _sheet = read_sample_sheet(sample_sheet)
-        sample_dict = extract_samples_from_sample_sheet(_sheet, input_dir)
+    # Check to see if running with test data. If so, fill in defaults
+    # for relevant empty parameters.
+    if test:
+        input_dir, sample_sheet, params, envs, output_dir = _setup_test()
+    elif not output_dir:
+        raise IOError("Must provide output directory to run.")
```

Comment:
Shall we uniformize double quotes into single quotes?

Reply:
👍